File: scripts/check_points_transactions.js
#!/usr/bin/env node
const { Client } = require('pg');

async function main() {
  const dbUrl = process.env.DATABASE_URL;
  if (!dbUrl) {
    console.error('Please set DATABASE_URL in your environment.');
    process.exit(2);
  }

  const client = new Client({ connectionString: dbUrl });
  try {
    await client.connect();
    const res = await client.query(
      `SELECT column_name, data_type, is_nullable
       FROM information_schema.columns
       WHERE table_name = 'points_transactions'
       ORDER BY ordinal_position;`
    );
    console.log('Columns for points_transactions:');
    console.table(res.rows);
  } catch (err) {
    console.error('Error querying database:', err.message || err);
    process.exit(1);
  } finally {
    await client.end();
  }
}

main();

File: scripts/check_points_transactions.py
#!/usr/bin/env python3
import os
import sys
import urllib.parse

try:
    import psycopg2
except Exception:
    print('psycopg2 is required. Install with: pip install psycopg2-binary')
    sys.exit(2)


def main():
    db_url = os.environ.get('DATABASE_URL')
    if not db_url:
        print('Please set DATABASE_URL in the environment.')
        sys.exit(2)

    # Parse to support libpq URI
    try:
        conn = psycopg2.connect(dsn=db_url)
    except Exception as e:
        print('Failed to connect to database:', e)
        sys.exit(1)

    cur = conn.cursor()
    try:
        cur.execute("""
            SELECT column_name, data_type, is_nullable
            FROM information_schema.columns
            WHERE table_name = 'points_transactions'
            ORDER BY ordinal_position;
        """)
        rows = cur.fetchall()
        if not rows:
            print('No columns found for points_transactions (table may not exist)')
        else:
            print('Columns for points_transactions:')
            for col, dtype, nullable in rows:
                print(f"- {col:25} {dtype:15} nullable={nullable}")
    except Exception as e:
        print('Error querying information_schema:', e)
        sys.exit(1)
    finally:
        cur.close()
        conn.close()


if __name__ == '__main__':
    main()
